runtime-modules/content/src/nft/types.rs

```diff
@@ -0,0 +1,329 @@
+use super::*;
+
+/// Metadata for NFT issuance
+pub type Metadata = Vec<u8>;
+
+pub type CuratorGroupId<T> = <T as ContentActorAuthenticator>::CuratorGroupId;
+pub type CuratorId<T> = <T as ContentActorAuthenticator>::CuratorId;
+pub type MemberId<T> = <T as membership::Trait>::MemberId;
+
+/// Owner royalty
+pub type Royalty = Perbill;
+
+/// NFT transactional status
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum TransactionalStatus<
+    BlockNumber: BaseArithmetic + Copy,
+    MemberId: Default + Copy + Ord,
+    Balance: Default,
+> {
+    Idle,
+    InitiatedOfferToMember(MemberId, Option<Balance>),
+    Auction(AuctionRecord<BlockNumber, Balance, MemberId>),
+    BuyNow(Balance),
+}
+
+impl<BlockNumber: BaseArithmetic + Copy, MemberId: Default + Copy + Ord, Balance: Default> Default
+    for TransactionalStatus<BlockNumber, MemberId, Balance>
+{
+    fn default() -> Self {
+        Self::Idle
+    }
+}
+
+/// Owned NFT representation
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct OwnedNFT<
+    BlockNumber: BaseArithmetic + Copy,
+    MemberId: Default + Copy + Ord,
+    Balance: Default,
+> {
+    pub owner: NFTOwner<MemberId>,
+    pub transactional_status: TransactionalStatus<BlockNumber, MemberId, Balance>,
+    pub creator_royalty: Option<Royalty>,
+}
+
+impl<
+        BlockNumber: BaseArithmetic + Copy,
+        MemberId: Default + Copy + PartialEq + Ord,
+        Balance: Default,
+    > OwnedNFT<BlockNumber, MemberId, Balance>
+{
+    /// Whether provided owner is nft owner
+    pub fn is_owner(&self, owner: &NFTOwner<MemberId>) -> bool {
+        self.owner.eq(owner)
+    }
+
+    /// Create new NFT
+    pub fn new(owner: NFTOwner<MemberId>, creator_royalty: Option<Royalty>) -> Self {
+        Self {
+            owner,
+            transactional_status: TransactionalStatus::Idle,
+            creator_royalty,
+        }
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum NFTOwner<MemberId> {
+    ChannelOwner,
+    Member(MemberId),
+}
+
+impl<MemberId> Default for NFTOwner<MemberId> {
+    fn default() -> Self {
+        Self::ChannelOwner
+    }
+}
+
+/// Information on the auction being created.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct Bid<MemberId, BlockNumber: BaseArithmetic + Copy, Balance> {
+    pub bidder: MemberId,
+    pub amount: Balance,
+    pub time: BlockNumber,
```

Comment:
> I think this variable `time` is the block at which the bid is submitted? I spent a long time thinking it was the unlocking time for an open auction bid.
> 
> So lets do as follows
> 
> * rename to `made_at_block`.
> * Lets make two different bid types `OpenAuctionBid` and `EnglishAuctionBid`, where only the the former has a `made_at_block` field, as this is not relevant to English auctions. Then we put the last bid field inside a part of the auction that captures whether its english or open. That way we have the most crisp representation.

Renaming done 2bfd2cad99.

I agree that currently `made_at_block` value is only relevant for `Open` auctions. 
Though with auction extension period feature for `English` auctions https://github.com/Joystream/joystream/pull/2667 , `made_at_block` value would be relevant for both types of auctions, because we need to be able to know when last bid was made to be able to activate extension period when necessary.

Reply:
Ok, I see.